### PR TITLE
MAINT: Make use of cmake configure for version

### PR DESCRIPTION
**Pull request by Krzmbrzl**
Instead of having to change the version number and the build year in all
those files manually, we let cmake do the work by replacing the
respective values in templates.

TODO:
- [x] Fix ToDo from below
- [x] Check that MacOS packaging script is still working properly (use the correct plist files)

### CODE REVIEW

**Review by davidebeatrici — APPROVED** [review]

File: overlay_winx64/mumble_ol.rc.in [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#ifdef SNAPSHOT_BUILD
+#define	VER_RELEASE	VS_FF_SPECIALBUILD|VS_FF_PRERELEASE
+#else
+#define VER_RELEASE 0L
+#endif
+
+VS_VERSION_INFO VERSIONINFO
+	FILEVERSION		@PROJECT_VERSION_MAJOR@,@PROJECT_VERSION_MINOR@,@PROJECT_VERSION_PATCH@,0
+	PRODUCTVERSION	@PROJECT_VERSION_MAJOR@,@PROJECT_VERSION_MINOR@,@PROJECT_VERSION_PATCH@,0
+	FILEFLAGSMASK	VS_FFI_FILEFLAGSMASK
```
Comment (Krzmbrzl) [inline overlay_winx64/mumble_ol.rc.in:23]: TODO: Fix alignment

File: overlay_winx64/overlay_exe_winx64/overlay_exe.rc.in [hunk 2]
```diff
@@ -35,7 +35,7 @@ VS_VERSION_INFO VERSIONINFO
 				VALUE "FileVersion", "@PROJECT_VERSION@"
 				VALUE "ProductVersion", "@PROJECT_VERSION@"
 				VALUE "LegalCopyright", "Copyright (c) 2005-2020 The Mumble Developers"
-				VALUE "OriginalFilename", "@OUTPUT_NAME@"
+				VALUE "OriginalFilename", "mumble_ol_helper_x64.exe"
```
Comment (Krzmbrzl) [inline overlay_winx64/overlay_exe_winx64/overlay_exe.rc.in:38]: TODO: Change back

File: g15helper/g15helper.plist.in [hunk 3]
```diff
@@ -13,7 +13,7 @@
 	<key>CFBundleSignature</key>
 	<string>G15H</string>
 	<key>CFBundleVersion</key>
-	<string>1.4.0</string>
+	<string>@PROJECT_VERSION_MAJOR@.@PROJECT_VERSION_MINOR@.@PROJECT_VERSION_PATCH@</string>
 	<key>NSHumanReadableCopyright</key>
 	<string>Copyright (c) 2005-2020 The Mumble Developers</string>
```
Comment (Krzmbrzl) [inline g15helper/g15helper.plist.in:18]: TODO: Replace 2020

**Comment by Krzmbrzl** [comment]
@davidebeatrici where's this error coming from:
```
CMake Error at C:/hostedtoolcache/windows/MumbleBuild/win64-static-1.4.x-2020-07-22-dbd6271-1162/scripts/buildsystems/vcpkg.cmake:331 (_add_executable):
  Cannot find source file:

    D:/a/1/b/g15helper/D:/a/1/b/g15helper/g15helper.rc
```
Does the build env use the `G15HELPER_RC` variable as well? Did you add that manually?

**Pull request merged**